File: scala-server/app/com/bolour/eagerwords/common/domain/UserGameSettings.scala
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/eagerwords.com/blob/master/LICENSE.md
 */

package com.bolour.eagerwords.common.domain

import com.bolour.eagerwords.common.domain.DeviceType.DeviceType
import com.bolour.eagerwords.common.domain.PieceProviderType.PieceProviderType
import com.bolour.eagerwords.common.domain.PlayerType.PlayerType

/**
  * User's game settings.
  *
  * @param playSettings: Settable preference parameters for starting games.
  * @param lookAndFeelSettings: Settable parameters affecting the look and feel of games.
  */
case class UserGameSettings(
  playSettings: GamePlayParams,
  lookAndFeelSettings: GameLookAndFeelParams
)

File: react-client/eagerwords/src/game/domain/UserGameSettings.js
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/eagerwords.com/blob/master/LICENSE.md
 */

import {defaultGamePlayParams} from "./GamePlayParams";
import {defaultGameLookAndFeelParams} from "./GameLookAndFeelParams";

export const mkUserGameSettings = function(playSettings, lookAndFeelSettings) {
  return {
    playSettings,
    lookAndFeelSettings
  }
};

export const defaultUserGameSettings =
  mkUserGameSettings(defaultGamePlayParams, defaultGameLookAndFeelParams);